ui/ui-components/components/schedule/RunAllSchedulesButton.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+import { ButtonProps } from "react-bootstrap";
+import { ApiHook } from "../../hooks/useApi";
+import { Actions } from "../../utils/apiData";
+import { successHandler } from "../../utils/eventHandlers";
+import LoadingButton from "../LoadingButton";
+
+interface Props {
+  execApi: ApiHook["execApi"];
+  modelId: string;
+  disabled?: boolean;
+  size?: ButtonProps["size"];
+  onStart?: () => void;
+  onSuccess?: () => void;
+  onComplete?: () => void;
+}
+
+const RunAllSchedulesButton: React.FC<Props> = ({
+  execApi,
+  modelId,
+  disabled,
+  size,
+  onStart,
+  onSuccess,
+  onComplete,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  async function enqueueAllSchedulesRun() {
+    onStart?.();
+    setIsLoading(true);
+    try {
+      const { runs } = await execApi<Actions.SchedulesRun>(
+        "post",
+        `/schedules/run`,
+        { modelId }
+      );
+      successHandler.set({ message: `${runs.length} runs enqueued` });
+      onSuccess?.();
+    } finally {
+      setIsLoading(false);
+      onComplete?.();
+    }
+  }
+
+  return (
+    <LoadingButton
+      variant="outline-primary"
+      size={size}
+      disabled={isLoading || disabled}
+      onClick={() => enqueueAllSchedulesRun()}
+      hideOn={["config"]}
+    >
+      Run all Schedules
+    </LoadingButton>
+  );
+};
+
+export default RunAllSchedulesButton;
```

Comment:
This introduces a new pattern into the ui-components of what I call "smart components" in that they also contain their own logic allowing things like this button to be shared in multiple pages (sources list and the model overview)

Reply:
We should avoid default exports outside of Pages as it makes refactor actions far more difficult.